src/data/adHocFilter.ts

```diff
@@ -26,23 +26,33 @@ export class AdHocFilter {
     if (this._targetTable === '' || !sql.match(new RegExp(`.*\\b${this._targetTable}\\b.*`, 'gi'))) {
       return sql;
     }
+
     let filters = adHocFilters
+      .filter(validate)
       .map((f, i) => {
         const key = f.key.includes('.') ? f.key.split('.')[1] : f.key;
         const value = isNaN(Number(f.value)) ? `\\'${f.value}\\'` : Number(f.value);
         const condition = i !== adHocFilters.length - 1 ? (f.condition ? f.condition : 'AND') : '';
         return ` ${key} ${f.operator} ${value} ${condition}`;
       })
       .join('');
+
+    if(filters === '') {
+      return sql;
+    }
     // Semicolons are not required and cause problems when building the SQL
     sql = sql.replace(';', '');
     return `${sql} settings additional_table_filters={'${this._targetTable}' : '${filters}'}`;
   }
 }
 
+function validate(filter: AdHocVariableFilter): boolean {
```

Comment:
Should we log in case of invalid filters?

Reply:
A log would be nice just for knowing how many filters aren't valid and potentially debugging issues.
Would notification of an invalid filter be helpful to the user? I'm not sure it would be but if it is we should figure out how to surface that.

Reply:
> Would notification of an invalid filter be helpful to the user?

What is the preferred way to create a notification in Grafana? Let's start with adding logs, and we can add another channel later.

Reply:
We could do something like [this](https://developers.grafana.com/ui/latest/index.html?path=/story/overlays-alert-toast--examples). But I would only want to do that if we can give the user a clear action to take based on that notification.